### ISSUE TITLE
Support for @NotNull is broken since version 2.4.0

### ISSUE THREAD

**Issue by egemenvardar** [stoicflame/enunciate#471]
Follow-up of #310.

This issue was fixed in 2.3.0. I recently updated my Enunciate version to 2.5.0 and realized that all input fields are again shown as optional. I tried the 2.4.0 version and the same problem happens there as well. Some changes in 2.4.0 must have broken this fix.

Could you please look into this? Thanks.

**Comment by stoicflame** [comment]
I've taken a look and can't seem to reproduce it. It's working for me.

If you'd like to provide me more details about how to reproduce the problem, I'd be happy to re-open this.

**Comment by egemenvardar** [comment]
@stoicflame I am using the `@NotNull` annotation from `javax.validation.constraints` to control the mandatory / optional fields.

This used to work fine with version 2.3.0. However, with versions 2.4.0 and 2.5.0 all fields are shown as optional; e.g.:

![enunciate_all_params_optional](https://cloud.githubusercontent.com/assets/17125072/17853210/68e53930-686b-11e6-9a2b-2f4e9d964c9b.png)

Although some of those fields are annotated with `@NotNull` they are still shown as optional. Note that if I just change my Enunciate version to 2.3.0 and keep everything else the same it does work. E.g. 

![enunciate_all_params_not_optional](https://cloud.githubusercontent.com/assets/17125072/17853298/da88c03e-686b-11e6-9311-afd1ada56ef3.png)

I hope this makes it more clear. Please let me know if you need more details.

**Comment by stoicflame** [comment]
I think I understand what you're reporting. I don't understand how to reproduce it.

Here's a snippet of code that I'm using to test it:

![enunciate-471-snippet-not-null](https://cloud.githubusercontent.com/assets/145838/17863276/47463cc6-6856-11e6-82a2-403dae67099c.png)

And here's the swagger UI that it produces (note that it's not optional):

![enunciate-471-swagger-not-null](https://cloud.githubusercontent.com/assets/145838/17863374/a63f5334-6856-11e6-957b-682002db2b34.png)

If I remove the `@NotNull`:

![enunciate-471-snippet-not-not-null](https://cloud.githubusercontent.com/assets/145838/17863429/d5bb29f8-6856-11e6-86ea-3cf493e3f9bd.png)

Then the Swagger UI shows it as optional:

![enunciate-471-swagger-not-not-null](https://cloud.githubusercontent.com/assets/145838/17863496/11b38ad6-6857-11e6-9cd3-53865ed78de2.png)

So I don't need more details about the problem itself; I need more details about how to reproduce the problem.

**Comment by egemenvardar** [comment]
Okay I see where the difference is. My getters and setters are generated by Lombok, e.g.

```
@Getter
@Setter
public class CreateReservationInput extends JsonLangInput {
	
	/** Identifier of the reservation offer */
	@JsonProperty("reservation_offer_id")
	@NotNull
	private Integer reservationOfferId;
	
	/** <code>true</code> if new reservation should not be allowed when the customer has an overlapping
	 * reservation, <code>false</code> otherwise */
	@JsonProperty("ignore_overlaps")
	@NotNull
	private Boolean ignoreOverlaps;
	
	/** <code>true</code> if the service should check whether the customer has a valid license;
	 * <code>false</code> otherwise */
	@JsonProperty("ignore_res_accs")
	@NotNull
	private Boolean ignoreResAccs;
...
}
```

My Lombok version 1.16.10. Note that if I don't use Lombok and write the getters and setters myself just like in your example it also works for me.

I always used Lombok for boilerplate code, and optional / mandatory fields used to work in Enunciate 2.3.0. Does that narrow down the problem?

**Comment by stoicflame** [comment]
[Enunciate 2.6 has been released.](https://github.com/stoicflame/enunciate/releases/tag/v2.6.0)

I set up a sample project as [described here](https://github.com/stoicflame/enunciate/wiki/Lombok), annotated the fields with `@NotNull` and verified that it works. There were a few adjustments to 2.6.0 that may fix the issue for you, so try it out. If it doesn't work, I'm probably going to need more details about how I can set up a sample project that exposes the problem.
